// documentation-loader/dist/documentation-loader.js
'use strict';

var _child_process = require('child_process');

module.exports = function scsslintLoader(source, map) {
    var args = [];
    var documentationConfig = [];
    var _cwd = process.cwd();

    function configCWD() {
        args.push('node ./node_modules/documentation/bin/documentation.js build');
        args.push(documentationConfig.entry);
        args.push('--output=' + documentationConfig.output);
        args.push('--format=' + documentationConfig.format);

        if (documentationConfig.theme) {
            args.push('--theme=' + documentationConfig.theme);
        }

        if (documentationConfig.shallow) {
            args.push('--shallow');
        }

        if (documentationConfig.github) {
            args.push('--github');
        }
        (0, _child_process.exec)(args.join(' '), {
            cwd: _cwd
        }, function (error) {
            if (error) {
                // console.log('documentation-loader error: ', error);
            }
        });
    }

    var documentationDefaults = {
        output: './documentation',
        github: false,
        format: 'json',
        config: '',
        theme: false,
        shallow: false,
        entry: this.resourcePath
    };
    var globalSettings = this.options.documentation || {};
    // const loader_settings = loaderUtils.parseQuery(this.query) || {};
    documentationConfig = Object.assign(documentationDefaults, globalSettings);
    this.cacheable();
    if (documentationConfig.entry.indexOf(_cwd) === 0) {
        documentationConfig.entry = documentationConfig.entry.substr(_cwd.length + 1);
    }
    configCWD();
    this.callback(null, source, map);
};
